build.rs

```diff
@@ -17,6 +17,8 @@ fn main() {
 
     if !compiler.nightly {
         println!("cargo:rustc-cfg=syn_disable_nightly_tests");
+    } else {
+        println!("cargo:rustc-cfg=syn_enable_doc_cfg");
```

Comment:
I think it'd be better to not use nightly detection to turn on this config, but rather explicitly enable it when running under docs.rs.

This can be done by removing the line here, and adding it to Cargo.toml, like how `tokio` does: https://github.com/tokio-rs/tokio/blob/0e729aa341028c121b9c39fe552ed4309bae6b6a/tokio/Cargo.toml#L132

Reply:
Ah, the metadata-field was added previously, but was lost during the second iteration and the force-push. I'll add it.

Nevertheless I'd keep the nightly-config, since this allows users to build the documentation locally with nightly and still getting the information.

Reply:
I don't think this is the right decision, as we don't want to automatically depend on nightly features on users computers, as that could break at any time. If `doc(cfg(...))` changes in the future, we would break nightly compilers building crates depending on this version of `syn`.

In general, we don't want to automatically enable unstable rust features on nightly compilers.

Reply:
Okay, seems very reasonable. I'll remove the feature from the build script and add it to the manifest.